sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -125,8 +127,27 @@ void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
     switch (Kind) {
     case alloc::device: {
       Id = detail::getSyclObjImpl(Dev)->getHandleRef();
-      Error = Plugin.call_nocheck<PiApiKind::piextUSMDeviceAlloc>(
-          &RetVal, C, Id, nullptr, Size, Alignment);
+      // Parse out buffer location property
+      // Buffer location is only supported on FPGA devices
+      bool IsBufferLocSupported = true;
+      if (!Dev.is_accelerator()) {
+        IsBufferLocSupported = false;
+      }
+      if (PropList.has_property<cl::sycl::ext::intel::experimental::property::
+                                    usm::buffer_location>() &&
+          IsBufferLocSupported) {
```

Comment:
```suggestion
      if (IsBufferLocSupported &&
          PropList.has_property<cl::sycl::ext::intel::experimental::property::
                                    usm::buffer_location>()) {
```
Checking `IsBufferLocSupported` first is cheap and can save us the additional check to `has_property`.